include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -559,23 +585,61 @@ bool
 Parameters::viscous_problem() const
 {
   return (equation_type == EquationType::Stokes || equation_type == EquationType::NavierStokes ||
-          use_turbulence_model == true);
+          turbulence_model_data.is_active);
+}
+
+bool
+Parameters::viscous_term_is_linear() const
+{
+  return not viscous_term_is_nonlinear();
+}
+
+bool
+Parameters::viscous_term_is_nonlinear() const
+{
+  return (viscosity_is_variable() &&
+          treatment_of_variable_viscosity == TreatmentOfVariableViscosity::Implicit);
+}
+
+bool
+Parameters::viscosity_is_variable() const
+{
+  return turbulence_model_data.is_active || generalized_newtonian_model_data.is_active;
+}
+
+bool
+Parameters::implicit_convective_problem() const
+{
+  if(convective_problem())
+  {
+    if(solver_type == SolverType::Steady ||
+       (solver_type == SolverType::Unsteady &&
+        (treatment_of_convective_term == TreatmentOfConvectiveTerm::Implicit)))
+    {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool
+Parameters::nonlinear_viscous_problem() const
+{
+  return viscous_problem() && viscous_term_is_nonlinear();
 }
 
 bool
 Parameters::nonlinear_problem_has_to_be_solved() const
 {
-  return convective_problem() &&
-         (solver_type == SolverType::Steady ||
-          (solver_type == SolverType::Unsteady &&
-           treatment_of_convective_term == TreatmentOfConvectiveTerm::Implicit));
+  // nonlinear_viscous_problem() ignored, i.e., does not trigger a Newton solve
+  return implicit_convective_problem();
```

Comment:
`linear_problem_has_to_be_solved` and `nonlinear_problem_has_to_be_solved` are used to deduce whether or not we execute  a Newton solver. For now, disable the nonlinear viscous term triggering a Newton solve.